service/src/test/kotlin/no/nav/su/se/bakover/service/søknadsbehandling/SøknadsbehandlingServiceIverksettTest.kt

```diff
@@ -190,104 +125,51 @@ internal class SøknadsbehandlingServiceIverksettTest {
-                .leggTilNyAttestering(Attestering.Iverksatt(attestant, attesteringstidspunkt)),
-            fritekstTilBrev = "",
-            stønadsperiode = behandling.stønadsperiode,
-            grunnlagsdata = Grunnlagsdata.IkkeVurdert,
-            vilkårsvurderinger = Vilkårsvurderinger.Søknadsbehandling.IkkeVurdert,
-        )
+        val expected = søknadsbehandlingIverksattInnvilget().second.copy(id = behandling.id)
 
         response shouldBe expected.right()
 
-        verify(søknadsbehandlingRepoMock).hent(behandling.id)
-        verify(utbetalingServiceMock).genererUtbetalingsRequest(
-            sakId = argThat { it shouldBe sakId },
-            attestant = argThat { it shouldBe attestant },
-            beregning = argThat { it shouldBe TestBeregning },
-            simulering = argThat { it shouldBe simulering },
-            uføregrunnlag = argThat { it shouldBe emptyList() },
+        verify(serviceAndMocks.søknadsbehandlingRepo).hent(behandling.id)
+        verify(serviceAndMocks.utbetalingService).genererUtbetalingsRequest(any(), any(), any(), any(), any())
+        verify(serviceAndMocks.søknadsbehandlingRepo).lagre(eq(expected), anyOrNull())
+        verify(serviceAndMocks.vedtakRepo).lagre(
+            argThat { it should beOfType<VedtakSomKanRevurderes.EndringIYtelse.InnvilgetSøknadsbehandling>() },
+            anyOrNull()
         )
-        verify(søknadsbehandlingRepoMock).lagre(eq(expected), anyOrNull())
-        var vedtak: VedtakSomKanRevurderes.EndringIYtelse.InnvilgetSøknadsbehandling? = null
-        verify(vedtakRepoMock).lagre(
-            argThat {
-                it should beOfType<VedtakSomKanRevurderes.EndringIYtelse.InnvilgetSøknadsbehandling>()
-                vedtak = it as VedtakSomKanRevurderes.EndringIYtelse.InnvilgetSøknadsbehandling
-                it.behandling shouldBe expected
-            },
+        verify(serviceAndMocks.utbetalingService).lagreUtbetaling(any(), anyOrNull())
+        verify(serviceAndMocks.kontrollsamtaleService).opprettPlanlagtKontrollsamtale(
+            argThat { it should beOfType<VedtakSomKanRevurderes.EndringIYtelse.InnvilgetSøknadsbehandling>() },
             anyOrNull()
         )
-        verify(utbetalingServiceMock).lagreUtbetaling(any(), anyOrNull())
-        verify(kontrollsamtaleServiceMock).opprettPlanlagtKontrollsamtale(argThat { it shouldBe vedtak }, anyOrNull())
-        verify(utbetalingServiceMock).publiserUtbetaling(argThat { it shouldBe simulertUtbetaling })
-        verify(behandlingMetricsMock).incrementInnvilgetCounter(BehandlingMetrics.InnvilgetHandlinger.PERSISTERT)
-        verify(statistikkObserver, times(2)).handle(any())
+        verify(serviceAndMocks.utbetalingService).publiserUtbetaling(argThat { it shouldBe simulertUtbetaling })
+        verify(serviceAndMocks.behandlingMetrics).incrementInnvilgetCounter(BehandlingMetrics.InnvilgetHandlinger.PERSISTERT)
+        verify(serviceAndMocks.observer, times(2)).handle(any())
```

Comment:
Hvorfor blir egentlig denne kalt to ganger?

Reply:
Fordi koden er som følger
```
iverksattBehandling.also {
                        observers.forEach { observer ->
                            observer.handle(
                                Event.Statistikk.SøknadsbehandlingStatistikk.SøknadsbehandlingIverksatt(
                                    iverksattBehandling,
                                ),
                            )
                            observer.handle((Event.Statistikk.Vedtaksstatistikk(vedtak)))
                        }
                    }
```
Mao, gjøres to statistikkoppføringer

Reply:
Okei, i ettertid tror jeg at jeg hadde favorisert et event-pattern over et observer-pattern. Men det ville fremdeles blitt 2 verifies .)

Reply:
Trenger vi egentlig teste at dette skjer? Det er ikke viktig for iverksettingen per se, så tenker at hvis vi ønsker å ha tester på at vi pusher til statistikk så burde det være egne tester som kun handler om statistikk: "iverksetting av søknadsbehandling pusher melding(er) til statistikk".